Sources/MongoSwift/BSON/Document+Codable.swift

```diff
@@ -46,7 +46,7 @@ extension Document: Codable {
         for key in container.allKeys {
             let k = key.stringValue
             if try container.decodeNil(forKey: key) {
-                try output.setValue(for: k, to: nil)
+                try output.setValue(for: k, to: NSNull())
```

Comment:
Is it still possible for `container.decode(AnyBSONValue.self, forKey: key)` below to yield an NSNull? If so, is this just necessary to catch actual `nil` values that might be in the container?

Reply:
Yeah, this is because now `AnyBSONValue` can represent a true null via `NSNull`. This is for that case.